binding/form.go

```diff
@@ -53,13 +57,30 @@ func (formMultipartBinding) Bind(req *http.Request, obj interface{}) error {
 	if err := req.ParseMultipartForm(defaultMemory); err != nil {
 		return err
 	}
-	if err := mapForm(obj, req.MultipartForm.Value); err != nil {
+	if err := mappingByPtr(obj, (*multipartRequest)(req), "form"); err != nil {
 		return err
 	}
 
-	if err := mapFiles(obj, req); err != nil {
-		return err
+	return validate(obj)
+}
+
+type multipartRequest http.Request
+
+var (
+	multipartFileHeaderStructType = reflect.TypeOf(multipart.FileHeader{})
+)
+
+func (r *multipartRequest) Set(value reflect.Value, field reflect.StructField, key string, opt setOptions) (isSetted bool, err error) {
```

Comment:
check: `var _ setter = &multipartRequest`

Reply:
What do you mean when you talk about "node"?
Btw, this method is not exported because the type `multipartRequest` is not exported, and I see no warnings about this from any linters.

Reply:
@vkd sorry, word typo, `note`.
Yes, it's not exported. We should add some usage note for new and important interface/method/class even if un-exported. thanks!

Reply:
Ok, thanks. I've added a couple of comments.